src/main/osd/osd_elements.c

```diff
@@ -779,13 +779,14 @@ static void osdElementGpsHomeDistance(osdElementParms_t *element)
 {
     if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
         const int32_t distance = osdGetMetersToSelectedUnit(GPS_distanceToHome);
-        tfp_sprintf(element->buff, "%d%c", distance, osdGetMetersToSelectedUnitSymbol());
+        tfp_sprintf(element->buff, "%c%d%c", SYM_HOMEFLAG, distance, osdGetMetersToSelectedUnitSymbol());
     } else {
+        element->buff[0] = SYM_HOMEFLAG;
         // We use this symbol when we don't have a FIX
-        element->buff[0] = SYM_COLON;
+        element->buff[1] = SYM_COLON;
         // overwrite any previous distance with blanks
-        memset(element->buff + 1, SYM_BLANK, 6);
-        element->buff[7] = '\0';
+        memset(element->buff + 2, SYM_BLANK, 6);
+        element->buff[8] = '\0';
```

Comment:
Not directly because of your changes, but this can be simplified as the original code unnecessarily does `// overwrite any previous distance with blanks`. It's not necessary to "clear" the field as the entire screen is cleared before any elements are drawn. So it's only necessary to draw what actually needs to be displayed - not clear possible previous data.

Reply:
Then it has changed:
- In the old OSD code, the screen was cleared between being drawn. 
- Later it was changed to let each element erase it's own drawn.
- Did you change the behaviour in your latest big refactor of the OSD? I haven't look at it deeply.

If true, I will remove it, of course.

Reply:
No, I didn't change the behavior. It's been this way for quite a while. For every OSD cycle the local buffer is completely cleared and then the elements are drawn fresh into the local buffer. Then the local buffer is streamed to the MAX7456 by only sending the characters in the local buffer that differ from the MAX7456 buffer. See the `osdDrawElements()` function in osd.c.

Reply:
This must have changed, all the `osd_elements.c` is full of erasing older value, or padding with blanks to be sure it is always the same length, for example:

https://github.com/betaflight/betaflight/blob/5afcb0985c04409b3d867679ae08b2b4707d5cbb/src/main/osd/osd_elements.c#L470-L471
https://github.com/betaflight/betaflight/blob/5afcb0985c04409b3d867679ae08b2b4707d5cbb/src/main/osd/osd_elements.c#L233-L236
https://github.com/betaflight/betaflight/blob/5afcb0985c04409b3d867679ae08b2b4707d5cbb/src/main/osd/osd_elements.c#L937-L939
...

So I think is better to let it as is, and remove all of them in other PR, if needed.

Reply:
It's just some of the GPS elements that are unnecessarily erasing the unused space (at the end). Probably all done by the same person and just wasn't caught in review. Adding padding to right-align text is fine, it's just the unnecessary attempt to "erase" parts of the screen that aren't needed.

Reply:
Just simplify it to:
```
static void osdElementGpsHomeDistance(osdElementParms_t *element)
{
    if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
        const int32_t distance = osdGetMetersToSelectedUnit(GPS_distanceToHome);
        tfp_sprintf(element->buff, "%c%d%c", SYM_HOMEFLAG, distance, osdGetMetersToSelectedUnitSymbol());
    } else {
        element->buff[0] = SYM_HOMEFLAG;
        // We use this symbol when we don't have a FIX
        element->buff[1] = SYM_COLON;
        element->buff[2] = '\0';
    }
}
```
and we're good. I'll look at fixing any others in a separate PR.

Reply:
I will change it tomorrow, I'm out of home. But I found the PR that changed the way the OSD cleared the elements, for curiosity: https://github.com/betaflight/betaflight/pull/4940

Reply:
And @mikeller needed to add the erasing of the warnings too: https://github.com/betaflight/betaflight/pull/5268/files so is clear that is not a bug that wasn't caught in the review. The OSD drawing must have changed at some moment in the latest months.

Reply:
@McGiverGim: I have a vague recollection of this, I think we _wanted_ to go the way of elements having to take care of this, and started down this way in #4940. But then at some point we found this wasn't really working, so we went for the more involved way that @etracer65 is describing. Some elements will still have the clearing up of unused space that was needed for the first approach.

Reply:
There's really no need for any element to "clear" space exactly. It's just that we use a top-left coordinate system for the positioning and a string for the element contents in most cases. So if the content is desired to display right-justified in any way we have to add leading padding to the string. The few other cases of where we're explicitly adding spies to clear the content **after** the actual content are unneeded and I'll take a look into cleaning these up in a separate PR.

Reply:
@etracer65 @mikeller it seems we discussed this before in other pr https://github.com/betaflight/betaflight/pull/5384 😁

Reply:
Yes, and I had the wrong approach then and have since fixed it. That was early in my understanding of the OSD code. The original code had the flaw of either drawing the text if disarmed, or drawing blanks if armed (again the "erasing" problem). My fix at the time was to only draw blanks one time when the state changed but that was misguided and not really needed. Now the code simply draws the "DISARMED" text when disarmed and nothing else.
```
static void osdElementDisarmed(osdElementParms_t *element)
{
    if (!ARMING_FLAG(ARMED)) {
        tfp_sprintf(element->buff, "DISARMED");
    }
}
```